core/src/main/java/nl/nn/adapterframework/filesystem/FileSystemActor.java

```diff
@@ -296,10 +298,19 @@ private String determineDestination(ParameterValueList pvl) throws FileSystemExc
 		throw new FileSystemException("no destination specified");
 	}
 
-	private F getFile(Message input, ParameterValueList pvl) throws FileSystemException {
+	private F getFile(@Nonnull Message input, ParameterValueList pvl) throws FileSystemException {
 		return fileSystem.toFile(determineFilename(input, pvl));
 	}
 
+	private F getFileAndCreateFolder(@Nonnull Message input, ParameterValueList pvl) throws FileSystemException {
+		String filenameWithFolder = determineFilename(input, pvl);
+		String folder = FilenameUtils.getFullPathNoEndSeparator(filenameWithFolder);
+		if(StringUtils.isNotBlank(folder) && !fileSystem.folderExists(folder) && isCreateFolder()) {
```

Comment:
Kan deze regel:
`StringUtils.isNotBlank(folder) && !fileSystem.folderExists(folder) && isCreateFolder()`

het volgende worden:

`StringUtils.isNotBlank(folder) &&  isCreateFolder() && !fileSystem.folderExists(folder) `

Op deze manier wordt een `fileSystem.folderExistst()` check voorkomen als we al weten dat we de folder toch niet willen maken.

Reply:
Ja slim!

Reply:
> Ja slim! (bij verschillende implemenaties zit in de createFolder een check om te kijken of de folder niet al bestaat)

Wordt er dan ook bij sommige implementaties de check `folderExists` dubbel uitgevoerd? 

Een klein puntje om rekening mee te houden, bij AWS kost elke interactie een heel klein beetje $ maar dat stapelt mogelijk wel op bij miljoenen berichtenn met extra onnodige interacties.

Reply:
Bij de copy en move zit het nu enorm knullig in elkaar...